pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -194,6 +203,33 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return sb.Unbind()
 	}
 
+	var emptyApplication v1alpha1.ApplicationSelector
+	if sbr.Spec.ApplicationSelector == emptyApplication ||
+		(sbr.Spec.ApplicationSelector.ResourceRef == "" &&
+			sbr.Spec.ApplicationSelector.LabelSelector.MatchLabels == nil) {
```

Comment:
I would also transform the above into a function to be used as predicate:

```go
if isApplicationSelectorEmpty(sbr.Spec.ApplicationSelector) {
    // ...
}
```

Reply:
Resolved 8383b4f